### ISSUE TITLE
Iterator case rule only updates for ... in ... declaration, but not the uses

### ISSUE THREAD

**Issue by StefanBalt** [jeremiah-c-leary/vhdl-style-guide#1561]
### Summary

The `parameter_specification_500` rule updates the iterator identifier in a `for ... in ...` declaration, but it does not update or report uses of that iterator later in the loop/generate body.

I am not sure whether this is intended behavior or a missing rule, but it leads to inconsistent casing after `--fix`.

### Current behavior

With this configuration:

```yaml
rule:
  parameter_specification_500:
    case: 'upper'
    disable: false
```

VSG changes the iterator declaration to uppercase but it does not change the iterator uses:

```VHDL
GEN_LABEL : for I in t_index_range generate
    I_INSTANCE : entity work.some_entity
        port map (
            CLK_I  => CLK_I,
            DATA_I => INPUT_DATA(i),
            DATA_O => OUTPUT_DATA(i)
        );
end generate GEN_LABEL;
```

So after --fix, the code ends up with mixed casing: declaration uses `I`, references still use `i`.

### Describe the solution you'd like
One of these, depending on intended design:

1. parameter_specification_500 should also report/fix uses of the iterator within the controlled loop/generate body, or
2. there should be a separate rule for iterator reference case consistency, or
3. the documentation should explicitly state that parameter_specification_500 only applies to the declaration token in the for header.

**Additional context**
This is related to the work from #1376 / PR #1382 that added parameter_specification_500.

My question is whether the current behavior is intentional scope, or whether iterator uses inside the body should also be checked.

**Comment by JHertz5** [comment]
Hi @StefanBalt 

Thanks for raising this issue. This is a good point. Of your suggested solutions, 2 is the option that fits best with the existing architecture of VSG. If you look at other similar rules, you can see that rules for capitalisation in the declaration (e.g. [signal_004](https://vhdl-style-guide.readthedocs.io/en/latest/signal_rules.html#signal-004)) are separate from the rules for consistent case (e.g. [signal_014](https://vhdl-style-guide.readthedocs.io/en/latest/signal_rules.html#signal-014)). So what we need in this case is a new rule to enforce consistent case for this type of token.

I'll take a look at whether I can implement a rule to resolve this.
